src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -526,3 +504,48 @@ def _check_n_components(n_components: Any) -> None:
     """Check that n_components is either 2, 4 or 6."""
     if n_components not in {2, 4, 6}:
         raise AttributeError(f'n_components must be either 2, 4 or 6 but is {n_components}')
+
+
+def _infer_n_components(frame: pl.DataFrame, column_specifiers: list[list[str]]) -> int | None:
+    """Infer number of components from DataFrame.
+
+    Method checks nested columns `pixel`, `position`, `velocity` and `acceleration` for number of
+    components by getting their list lenghts, which must be equal for all else a ValueError is
+    raised. Additionally, a list of list of column specifiers is checked for consistency.
+
+    Parameters
+    ----------
+    frame: pl.DataFrame
+        DataFrame to check.
+    column_specifiers:
+        List of list of column specifiers.
+
+    Returns
+    -------
+    int or None
+        Number of components
+
+    Raises
+    ------
+    ValueError
+        If number of components is not equal for all considered columns and rows.
+    """
+    all_considered_columns = ['pixel', 'position', 'velocity', 'acceleration']
```

Comment:
maybe we can put all_considered_columns in the arg list? maybe we extend the list of columns in the future, or someone wants to use additional (project) specific columns?
We can set the default to the predefined list of columns.

Reply:
Adding some kind of support for non-standard columns in the future is a good idea, but that's actually a feature request.
As that would be out of scope of this PR which is a bugfix, I have created a new issue for this were we can discuss details: #524

Reply:
cool. thx

Reply:
Also, keep in mind that setting lists (or any other mutable datatype) as default arguments in functions introduces horrible bugs in python:

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/